internal/services/cdn/cdn_frontdoor_rule_sets_association_resource.go

```diff
@@ -0,0 +1,223 @@
+			Read:   pluginsdk.DefaultTimeout(5 * time.Minute),
+			Delete: pluginsdk.DefaultTimeout(30 * time.Minute),
+		},
+
+		Importer: pluginsdk.ImporterValidatingResourceIdThen(func(id string) error {
+			_, err := parse.FrontDoorRuleSetAssociationID(id)
+			return err
+		}, func(ctx context.Context, d *pluginsdk.ResourceData, meta interface{}) (data []*pluginsdk.ResourceData, err error) {
+			id, err := parse.FrontDoorRuleSetAssociationID(d.Id())
+			if err != nil {
+				return []*pluginsdk.ResourceData{}, err
+			}
+
+			client := meta.(*clients.Client).Cdn.FrontDoorRoutesClient
+			resp, err := client.Get(ctx, id.ResourceGroup, id.ProfileName, id.AfdEndpointName, id.AssociationName)
+			if err != nil {
+				return []*pluginsdk.ResourceData{}, fmt.Errorf("retrieving %s: %+v", id, err)
+			}
+
+			if resp.RouteProperties == nil {
+				return []*pluginsdk.ResourceData{}, fmt.Errorf("retrieving %s: 'RouteProperties' are 'nil'", id)
+			}
+			ruleSets := flattenRuleSetResourceArray(resp.RouteProperties.RuleSets)
+
+			d.SetId(id.ID())
+			d.Set("cdn_frontdoor_route_id", parse.NewFrontDoorRouteID(id.SubscriptionId, id.ResourceGroup, id.ProfileName, id.AfdEndpointName, id.AssociationName).ID())
+			d.Set("cdn_frontdoor_rule_set_ids", ruleSets)
+
+			return []*pluginsdk.ResourceData{d}, nil
+		}),
+
+		Schema: map[string]*pluginsdk.Schema{
+			"cdn_frontdoor_route_id": {
+				Type:         pluginsdk.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: validate.FrontDoorRouteID,
+			},
+
+			"cdn_frontdoor_rule_set_ids": {
+				Type:     pluginsdk.TypeSet,
+				Required: true,
+				ForceNew: true,
+
+				Elem: &pluginsdk.Schema{
+					Type:         pluginsdk.TypeString,
+					ValidateFunc: validate.FrontDoorRuleSetID,
+				},
+			},
+		},
+	}
+}
+
+func resourceCdnFrontDoorRuleSetAssociationCreate(d *pluginsdk.ResourceData, meta interface{}) error {
+	client := meta.(*clients.Client).Cdn.FrontDoorRoutesClient
+	ctx, cancel := timeouts.ForCreate(meta.(*clients.Client).StopContext, d)
+	defer cancel()
+
+	routeId, err := parse.FrontDoorRouteID(d.Get("cdn_frontdoor_route_id").(string))
+	if err != nil {
+		return err
+	}
+
+	// NOTE: The association name is the name of the route the resource is being associated with...
+	// e.g. subscriptions/12345678-1234-9876-4563-123456789012/resourceGroups/resGroup1/providers/Microsoft.Cdn/profiles/profile1/afdEndpoints/endpoint1/associations/route1
+	id := parse.NewFrontDoorRuleSetAssociationID(routeId.SubscriptionId, routeId.ResourceGroup, routeId.ProfileName, routeId.AfdEndpointName, routeId.RouteName)
+
+	// lock the route, association and rule set resource types for create...
+	locks.ByID(cdnFrontDoorRouteResourceType)
+	defer locks.UnlockByID(cdnFrontDoorRouteResourceType)
+
+	locks.ByID(cdnFrontDoorRuleSetAssociationResourceType)
+	defer locks.UnlockByID(cdnFrontDoorRuleSetAssociationResourceType)
+
+	locks.ByID(cdnFrontDoorRuleSetResourceType)
```

Comment:
I'm still hesitant about locking based on resource type instead of the id of the resource but open to it with some additional context of why we're going that way!